src/repart/repart.c

```diff
@@ -5827,76 +5823,98 @@ static int add_subvolume_path(const char *path, Set **subvolumes) {
         if (r < 0)
                 return log_error_errno(r, "Failed to stat source file '%s/%s': %m", strempty(arg_copy_source), path);
 
-        r = set_ensure_consume(subvolumes, &inode_hash_ops, TAKE_PTR(st));
+        r = hashmap_ensure_put(subvolumes, &inode_hash_ops, st, UINT_TO_PTR(flags));
         if (r < 0)
                 return log_oom();
+        if (r == 0)
+                log_debug("Found duplicate subvolume path %s for the same partition", path);
```

Comment:
I think this also will never happen in practice because we'll only call this when iterating over a hashmap, so I'd drop it.